ssh/tailssh/tailssh.go

```diff
@@ -654,15 +722,15 @@ func matchRule(r *tailcfg.SSHRule, ci *sshConnInfo) (a *tailcfg.SSHAction, local
 	if r.RuleExpires != nil && ci.now.After(*r.RuleExpires) {
 		return nil, "", errRuleExpired
 	}
-	if !matchesPrincipal(r.Principals, ci) {
-		return nil, "", errPrincipalMatch
-	}
```

Comment:
why is this moving?

Reply:
I want to avoid fetching SSH certs in the case where the SSH user can't match anyway.

This moves it as late as possible.